kevm-pyk/src/kevm_pyk/kproj/evm-semantics/evm.md

```diff
@@ -1287,51 +1288,51 @@ Precompiled Contracts
+    rule #precompiled(6p256) => ECADD
+    rule #precompiled(7p256) => ECMUL
+    rule #precompiled(8p256) => ECPAIRING
+    rule #precompiled(9p256) => BLAKE2F
+    rule #precompiled(10p256) => KZGPOINTEVAL
+    rule #precompiled(11p256) => BLS12G1ADD
+    rule #precompiled(12p256) => BLS12G1MSM
+    rule #precompiled(13p256) => BLS12G2ADD
+    rule #precompiled(14p256) => BLS12G2MSM
+    rule #precompiled(15p256) => BLS12PAIRING_CHECK
+    rule #precompiled(16p256) => BLS12MAPFPTOG1
+    rule #precompiled(17p256) => BLS12MAPFP2TOG2
+
+    syntax MInt{256} ::= #precompiledAccountsUB ( Schedule ) [symbol(#precompiledAccountsUB), function, total]
  // ----------------------------------------------------------------------------------------------------
-    rule #precompiledAccountsUB(FRONTIER)          => 4
+    rule #precompiledAccountsUB(FRONTIER)          => 4p256
     rule #precompiledAccountsUB(HOMESTEAD)         => #precompiledAccountsUB(FRONTIER)
     rule #precompiledAccountsUB(TANGERINE_WHISTLE) => #precompiledAccountsUB(HOMESTEAD)
     rule #precompiledAccountsUB(SPURIOUS_DRAGON)   => #precompiledAccountsUB(TANGERINE_WHISTLE)
-    rule #precompiledAccountsUB(BYZANTIUM)         => 8
+    rule #precompiledAccountsUB(BYZANTIUM)         => 8p256
     rule #precompiledAccountsUB(CONSTANTINOPLE)    => #precompiledAccountsUB(BYZANTIUM)
     rule #precompiledAccountsUB(PETERSBURG)        => #precompiledAccountsUB(CONSTANTINOPLE)
-    rule #precompiledAccountsUB(ISTANBUL)          => 9
+    rule #precompiledAccountsUB(ISTANBUL)          => 9p256
     rule #precompiledAccountsUB(BERLIN)            => #precompiledAccountsUB(ISTANBUL)
     rule #precompiledAccountsUB(LONDON)            => #precompiledAccountsUB(BERLIN)
     rule #precompiledAccountsUB(MERGE)             => #precompiledAccountsUB(LONDON)
     rule #precompiledAccountsUB(SHANGHAI)          => #precompiledAccountsUB(MERGE)
-    rule #precompiledAccountsUB(CANCUN)            => 10
-    rule #precompiledAccountsUB(PRAGUE)            => 17
+    rule #precompiledAccountsUB(CANCUN)            => 10p256
+    rule #precompiledAccountsUB(PRAGUE)            => 17p256
 
 
     syntax Set ::= #precompiledAccountsSet    ( Schedule ) [symbol(#precompiledAccountsSet),    function, total]
-    syntax Set ::= #precompiledAccountsSetAux ( Int      ) [symbol(#precompiledAccountsSetAux), function, total]
+    syntax Set ::= #precompiledAccountsSetAux ( MInt{256}) [symbol(#precompiledAccountsSetAux), function, total]
  // ------------------------------------------------------------------------------------------------------------
     rule #precompiledAccountsSet(SCHED) => #precompiledAccountsSetAux(#precompiledAccountsUB(SCHED))
 
-    rule #precompiledAccountsSetAux(N)  => .Set requires N <=Int 0
-    rule #precompiledAccountsSetAux(N)  => SetItem(N) #precompiledAccountsSetAux(N -Int 1) [owise, preserves-definedness]
+    rule #precompiledAccountsSetAux(N)  => .Set requires N <=uMInt 0p256
```

Comment:
here you can just compare if it is equal to `0p256`. An unsigned mint cannot ever be negative.

Reply:
Done!